text/0035-schema-applies-to-unspecified.md

```diff
@@ -0,0 +1,90 @@
+# Change the Semantics `appliesTo` Fields in the Schema
+
+## Related issues and PRs
+
+- Reference Issues: [cedar#351](https://github.com/cedar-policy/cedar/issues/351)
+- Implementation PR(s):
+
+## Timeline
+
+- Started: 2023-10-25
+
+## Summary
+
+| Scenario | Status Quo | Proposal |
+| --------- | -------- | ------- |
+| Specified `appliesTo` but omitted `principalTypes`/`resourceTypes` | Authz request with this action can only have unspecified entity as principal/resource | Same as status quo |
+| Omitted `appliesTo` | Authz request with this action can only have unspecified entity as principal/resource | Action cannot be used in authz request |
+| `appliesTo.principalTypes: []` / `appliesTo.resourceTypes: []` | Action cannot be used in authz request | ❌ Not allowed |
+| `appliesTo` is defined but all sub-elements are omitted | Authz request with this action can only have unspecified entity as principal/resource; context is an empty record | ❌ Not allowed |
+
+## Motivation
+
+### Current behavior
+
+In Cedar 2.x, if an action's `appliesTo.principalTypes` or `appliesTo.resourceTypes` is not given (or if the entire `appliesTo` element is not given), then it's interpreted as an action that applies to only unspecified principals and resources.
+An _unspecified_ entity has a type that is distinct from any defined type.
+It is created by passing the `None` option in the principal and/or resource component of a [`Request`](https://docs.rs/cedar-policy/latest/cedar_policy/struct.Request.html).
+If an action's `appliesTo.principalTypes` or `appliesTo.resourceTypes` fields are set to the empty list, then the action is not expected to be used with any request; it should only be used as a way to group other actions.
```

Comment:
Worth adding: if you indicate that `principalTypes` is `[]` then it rules out using the action in requests _even if you don't specify `resourceTypes` as `[]`. This is confusing, and part of the motivation of this RFC is to address this source of confusion. (You say this below, under challenges, but it is a little strange to me to introduce the issue there for the first time. The first sentence in that part is recapping what was written above.)

Reply:
Revised text. Lmk what you think